### PR TITLE
Move from CRA-TS to CRA

### PR DESCRIPTION
**Pull request by maracuja-juice**
As the release for CRA3 is coming closer and closer I decided to start to tackle #129 using their newest beta.
Biggest downside I noticed so far is that they hide their eslint rules and you can't customize them but as they so far seem to be pretty reasonable that's ok.

On a positive note: It solved a long standing issue for me with the Jest watch mode.

Challenges presented along the way:
- "unknown browser kaios" -> delete node_modules and yarn.lock and do full reinstall
- "ReactSixteenAdapter is not a constructor" -> import without the asterix and use the enzyme-to-json serializer because of the new Jest 24 behaviour.

### CODE REVIEW

File: .storybook/webpack.config.js [hunk 1]
```diff
@@ -2,14 +2,7 @@ const path = require('path');
...
     config.module.rules.push({
-        test: /\.scss$/,
-        loaders: ["style-loader", "css-loader", "sass-loader"],
-        include: path.resolve(__dirname, '../src')
-    });
-    config.resolve.extensions.push('.scss');
-
-    config.module.rules.push({
-        test: /\.(ts|tsx)$/,
+        test: /(?:test)\.(ts|tsx)$/,
```
Comment (maracuja-juice) [inline .storybook/webpack.config.js]: If the described issue is fixed, I think this can be removed.

**Comment by RohrerF** [comment]
@maracuja-juice due to the newly used serializer the snapshots now look different. Are you happy with them or do you dislike this format

**Comment by maracuja-juice** [comment]
The new format makes a lot more sense than the old format. It now only records the actual output and not also some weird enzyme methods that we don't care about at all.
Good job for figuring this out! I searched in the wrong GitHub Repos.
(Gonna look at the other changes later)

File: src/setupTests.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { configure } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+
+configure({ adapter: new Adapter() });
```
Comment (maracuja-juice) [inline src/setupTests.ts]: From where is this setup called? 
Is this automatically called by Jest?

File: src/views/Analysis/__snapshots__/AnalysisGame.test.tsx.snap [hunk 3]
```diff
@@ -1,1209 +1,22 @@
...
-}
+  <div
+    className="analysisLayout"
+  >
+    <div
+      className="moveList"
+      id="moveList"
+    />
+    <div
+      className="historyButtons"
```
Comment (maracuja-juice) [inline src/views/Analysis/__snapshots__/AnalysisGame.test.tsx.snap]: This seems to have been an issue in the old snapshot as well but shouldn't our custom Components also be represented in this snapshot as a children of these divs? (they are directly used in the AnalysisGame component)

File: src/views/Game/__snapshots__/Game.test.tsx.snap [hunk 4]
```diff
@@ -1,7014 +1,994 @@
...
-    activeBoards={Array []}
-    board={
+<div
+  className="center"
+>
+  <div
+    className="gameWrapper"
+  >
+    <ConnectFunction />
+    <BigBoard
```
Comment (maracuja-juice) [inline src/views/Game/__snapshots__/Game.test.tsx.snap]: Reference to the other comment: Here there is the BigBoard represented as a child but not in the other snapshot 😲

File: .npmrc [hunk 5]
```diff
@@ -0,0 +1 @@
+registry=https://registry.yarnpkg.com/
```
Comment (maracuja-juice) [inline .npmrc]: Makes sense 👍

File: src/views/Analysis/__snapshots__/AnalysisGame.test.tsx.snap (hunk 3 above)
Comment (maracuja-juice) [reply src/views/Analysis/__snapshots__/AnalysisGame.test.tsx.snap]: Ah I figured it out. It doesn't have the moves so it doesn't render them.
In #169 I indirectly addressed this by decoupling the actual Analysis Display from the Routing / loading stuff.

File: src/views/Game/__snapshots__/Game.test.tsx.snap (hunk 4 above)
Comment (maracuja-juice) [reply src/views/Game/__snapshots__/Game.test.tsx.snap]: Not actually a problem with the snapshots as commented above.

Comment (maracuja-juice) [reply src/views/Game/__snapshots__/Game.test.tsx.snap]: Not actually a problem with the snapshots as commented above.

File: .storybook/webpack.config.js (hunk 1 above)
Comment (maracuja-juice) [reply .storybook/webpack.config.js]: Turns out that this was not the case. There was still no Typescript output from the story files when using the old configuration.
We don't need the files containing `.test` in Storybook anyway though so we don't have to find a solution around this.

**Comment by maracuja-juice** [comment]
As the current beta seems to be stable I don't see a problem with merging this and starting to work with the beta version.

File: src/setupTests.ts (hunk 2 above)
Comment (RohrerF) [reply src/setupTests.ts]: yes. Check out the [dokumentation](https://facebook.github.io/create-react-app/docs/running-tests#src-setuptestsjs)

**Pull request merged**